### PR TITLE
better display of backend metrics

### PR DESCRIPTION
**Pull request by Keksoj**
reconstruct a map of backend metrics to avoid listing a backend id several times.

Before:

```
MyCluster/lagging-server
--------
┌─────────────────────────┬───────┬───────┐
│                         │ gauge │ count │
├─────────────────────────┼───────┼───────┤
│ connections_per_backend │ 0     │       │
└─────────────────────────┴───────┴───────┘

MyCluster/lagging-server
--------
┌─────────────────┬───────┬───────┐
│                 │ gauge │ count │
├─────────────────┼───────┼───────┤
│ http.status.2xx │       │ 1167  │
└─────────────────┴───────┴───────┘

MyCluster/lagging-server
--------
┌──────────┬───────┬───────┐
│          │ gauge │ count │
├──────────┼───────┼───────┤
│ bytes_in │       │ 258   │
└──────────┴───────┴───────┘

MyCluster/lagging-server
--------
┌───────────┬───────┬───────┐
│           │ gauge │ count │
├───────────┼───────┼───────┤
│ bytes_out │       │ 699   │
└───────────┴───────┴───────┘

MyCluster/lagging-server
--------
┌───────────────────────┬─────────┬─────┬─────┬─────┬───────┬────────┬─────────┬──────┐
│ Percentiles           │ samples │ p50 │ p90 │ p99 │ p99.9 │ p99.99 │ p99.999 │ p100 │
├───────────────────────┼─────────┼─────┼─────┼─────┼───────┼────────┼─────────┼──────┤
│ backend_response_time │ 3       │ 0   │ 0   │ 0   │ 0     │ 0      │ 0       │ 0    │
└───────────────────────┴─────────┴─────┴─────┴─────┴───────┴────────┴─────────┴──────┘

MyCluster/lagging-server
--------
┌─────────────────────────┬─────────┬─────┬─────┬─────┬───────┬────────┬─────────┬──────┐
│ Percentiles             │ samples │ p50 │ p90 │ p99 │ p99.9 │ p99.99 │ p99.999 │ p100 │
├─────────────────────────┼─────────┼─────┼─────┼─────┼───────┼────────┼─────────┼──────┤
│ backend_connection_time │ 3       │ 0   │ 0   │ 0   │ 0     │ 0      │ 0       │ 0    │
└─────────────────────────┴─────────┴─────┴─────┴─────┴───────┴────────┴─────────┴──────┘

MyCluster/lagging-server
--------
┌──────────┬───────┬───────┐
│          │ gauge │ count │
├──────────┼───────┼───────┤
│ requests │       │ 3     │
└──────────┴───────┴───────┘

MyCluster/lagging-server
--------
┌───────────────────┬───────┬───────┐
│                   │ gauge │ count │
├───────────────────┼───────┼───────┤
│ access_logs.count │       │ 1167  │
└───────────────────┴───────┴───────┘
```

After:

```
MyCluster/lagging-server
--------
┌─────────────────────────┬───────┬───────┐
│                         │ gauge │ count │
├─────────────────────────┼───────┼───────┤
│ access_logs.count       │       │ 1167  │
├─────────────────────────┼───────┼───────┤
│ bytes_in                │       │ 258   │
├─────────────────────────┼───────┼───────┤
│ bytes_out               │       │ 699   │
├─────────────────────────┼───────┼───────┤
│ connections_per_backend │ 0     │       │
├─────────────────────────┼───────┼───────┤
│ http.status.2xx         │       │ 1167  │
├─────────────────────────┼───────┼───────┤
│ requests                │       │ 3     │
└─────────────────────────┴───────┴───────┘
┌─────────────────────────┬─────────┬─────┬─────┬─────┬───────┬────────┬─────────┬──────┐
│ Percentiles             │ samples │ p50 │ p90 │ p99 │ p99.9 │ p99.99 │ p99.999 │ p100 │
├─────────────────────────┼─────────┼─────┼─────┼─────┼───────┼────────┼─────────┼──────┤
│ backend_connection_time │ 3       │ 0   │ 0   │ 0   │ 0     │ 0      │ 0       │ 0    │
├─────────────────────────┼─────────┼─────┼─────┼─────┼───────┼────────┼─────────┼──────┤
│ backend_response_time   │ 3       │ 0   │ 0   │ 0   │ 0     │ 0      │ 0       │ 0    │
└─────────────────────────┴─────────┴─────┴─────┴─────┴───────┴────────┴─────────┴──────┘
```

### CODE REVIEW

File: command/src/proto/display.rs [hunk 1]
```diff
@@ -269,9 +269,24 @@ fn print_cluster_metrics(cluster_metrics: &BTreeMap<String, ClusterMetrics>) {
...
+                    .entry(backend.backend_id.clone())
+                    .and_modify(|map| {
+                        map.insert(metric_name.clone(), value.clone());
+                    })
+                    .or_insert(BTreeMap::from([(metric_name.clone(), value.clone())]));
+            }
+        }
+
+        for (backend_id, metrics) in backend_metric_acc {
+            println!("\n{cluster_id}/{}\n--------", backend_id);
```
Comment (llenotre) [inline command/src/proto/display.rs:288]:
```suggestion
            println!("\n{cluster_id}/{backend_id}\n--------");
```

File: command/src/proto/display.rs [hunk 2]
```diff
@@ -269,9 +269,24 @@ fn print_cluster_metrics(cluster_metrics: &BTreeMap<String, ClusterMetrics>) {
         print_gauges_and_counts(&filtered);
         print_percentiles(&filtered);
 
-        for backend_metrics in cluster_metrics_data.backends.iter() {
-            println!("\n{cluster_id}/{}\n--------", backend_metrics.backend_id);
-            let filtered = filter_metrics(&backend_metrics.metrics);
+        // backend_id -> (metric_name -> value )
+        let mut backend_metric_acc: BTreeMap<String, BTreeMap<String, FilteredMetrics>> =
```
Comment (llenotre) [inline command/src/proto/display.rs:273]: Use a `HashMap` instead?

File: command/src/proto/display.rs (hunk 1 above)
Comment (Keksoj) [reply command/src/proto/display.rs]: done!

File: command/src/proto/display.rs (hunk 2 above)
Comment (Keksoj) [reply command/src/proto/display.rs]: done!

**Review by Wonshtrum — APPROVED** [review]

**Pull request merged**